script.js

```diff
@@ -73,159 +73,201 @@ if (
     };
   });
 
-  //------ Rating filter -------
+  //--------- FILTER BY RATING ----------
   const ratingInput = document.getElementsByName("rating");
 
-  function ratingFilter() {
-    let cards = document.querySelectorAll(".challenge-item");
-    for (let i = 0; i < cards.length; i++) {
-      if (
-        cards[i].querySelector("ul.rating").ariaValueNow >=
-          current_star_level_from &&
-        cards[i].querySelector("ul.rating").ariaValueNow <=
-          current_star_level_to
-      ) {
-        cards[i].classList.remove("is-hidden");
-      } else {
-        cards[i].classList.add("is-hidden");
+  class RatingFilter {
+    constructor(list) {
+    }
+    challengeMatch(card) {
+      let cards = document.querySelectorAll(".challenge-item");
+      for (let i = 0; i < cards.length; i++) {
+        if (
+          card.querySelector("ul.rating").ariaValueNow >=
+            current_star_level_from &&
+          card.querySelector("ul.rating").ariaValueNow <=
+            current_star_level_to
```

Comment:
Storing data in the DOM like this is risky, because it couples your data with your display of that data. If someone wants to change the way data is displayed, they might change the DOM (e.g. no longer use `<ul>`). That would happen somewhere else in the code, but it would result in this part of the code breaking.

Having dependencies like this, where changes to one part of the code can have unforeseen consequences to another part of the code, is what we try to avoid with OOP.

Another way of doing this would be to have a card object type (i.e. a `Card` class or something) with properties for the rating value, which are used both here and when rendering the DOM elements. That makes the properties the source of truth, and more clearly communicates to programmers (because they are part of the public interface of the class) that it's risky to change them.